tests/e2e/steps_inactive_vals.go

```diff
@@ -0,0 +1,500 @@
+package main
+
+import (
+	"strconv"
+
+	gov "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+)
+
+// stepsInactiveValidatorsOnConsumer tests situations where validators that are *not* in the active set on the
+// provider chain validate on the consumer chain.
+// The provider chain is set to have at most *2* validators active in consensus, and there are 3 validators in total.
+// high-level, this test does:
+// - start the provider chain
+// - start a consumer chain
+// - check that non-consensus validators do not get slashed for downtime; and that they don't get rewards
+// - check that active validators *do* get slashed for downtime, and don't get rewards while they are down
+// - check that non-consensus validators *do* get jailed for downtime on consumer chains
+// - check that non-consensus validators *become* consensus validators when they have enough power
+func stepsInactiveProviderValidators() []Step {
+	s := concatSteps(
+		[]Step{
+			{
+				Action: StartChainAction{
+					Chain: ChainID("provi"),
+					Validators: []StartChainValidator{
+						{Id: ValidatorID("alice"), Stake: 100000000, Allocation: 10000000000},
+						{Id: ValidatorID("bob"), Stake: 200000000, Allocation: 10000000000},
+						{Id: ValidatorID("carol"), Stake: 300000000, Allocation: 10000000000},
+					},
+				},
+				State: State{
+					ChainID("provi"): ChainState{
+						ValPowers: &map[ValidatorID]uint{
+							ValidatorID("alice"): 0, // max consensus validators is 2, so alice should not be in power
+							ValidatorID("bob"):   200,
+							ValidatorID("carol"): 300,
+						},
+						StakedTokens: &map[ValidatorID]uint{
+							ValidatorID("alice"): 100000000,
+							ValidatorID("bob"):   200000000,
+							ValidatorID("carol"): 300000000,
+						},
+						Rewards: &Rewards{
+							IsNativeDenom:       true, // check for rewards in the provider denom
+							IsIncrementalReward: true, // check current rewards, because alice gets one block of rewards due to being in the genesis
```

Comment:
Not sure I fully understand the comment "because alice gets one block of rewards" because exactly below we have `IsRewarded` for alice to be `false`.

Reply:
adjusted: "we need to get incremental rewards. if we would look at total rewards, alice would trivially also get rewards, because she gets rewards in the first block due to being in the genesis."